EscapeHTML=false doesn't work

set EscapeHTML=false but the following test doesn't work:

```go
package main

import (
        "bytes"
        "encoding/json"
        "fmt"
        "testing"

        jsoniter "github.com/json-iterator/go"
)

var marshalConfig = jsoniter.Config{
        EscapeHTML:             false,
        SortMapKeys:            true,
        ValidateJsonRawMessage: true,
}.Froze()

type Foo struct {
        Bar interface{}
}

func (f *Foo) MarshalJSON() ([]byte, error) {
        ret, err := marshalConfig.Marshal(&f.Bar)
        fmt.Println("MarshalJSON ret", string(ret), err)
        return ret, err
}

func TestFooEncode(t *testing.T) {
        foo := &Foo{
                Bar: []string{"123<ab>", "ooo"},
        }
        out, err := marshalConfig.Marshal(foo)
        if err != nil {
                t.Error("err must be nil")
                return
        }

        var stdbuf bytes.Buffer
        stdenc := json.NewEncoder(&stdbuf)
        stdenc.SetEscapeHTML(false)
        stdenc.Encode(foo)

        stdout := string(stdbuf.Bytes())
        if stdout[len(stdout)-1:] == "\n" {
                stdout = stdout[:len(stdout)-1]
        }
        bufout := string(out)
        fmt.Println("stdout", stdout)
        fmt.Println("bufout", bufout)
        if stdout != bufout {
                t.Error(fmt.Sprintf("should equal,std=%v,buf=%v", stdout, bufout))
        }
}
```

the output is : 
```
MarshalJSON ret ["123<ab>","ooo"] <nil>
MarshalJSON ret ["123<ab>","ooo"] <nil>
stdout ["123<ab>","ooo"]
bufout ["123\u003cab\u003e","ooo"]
--- FAIL: TestFooEncode (0.00s)
    fail_test.go:52: should equal,std=["123<ab>","ooo"],buf=["123\u003cab\u003e","ooo"]
FAIL
exit status 1
```

we can see that the json was correctly marshaled in Foo.MarshalJSON, but then escaped again by standard json package?

i think pull request #337 may caused this issue.

--- a/api_tests/marshal_json_escape_test.go
+++ b/api_tests/marshal_json_escape_test.go
@@ -1,0 +1,47 @@
+package test
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/stretchr/testify/require"
+)
+
+var marshalConfig = jsoniter.Config{
+	EscapeHTML:             false,
+	SortMapKeys:            true,
+	ValidateJsonRawMessage: true,
+}.Froze()
+
+type Container struct {
+	Bar interface{}
+}
+
+func (c *Container) MarshalJSON() ([]byte, error) {
+	return marshalConfig.Marshal(&c.Bar)
+}
+
+func TestEncodeEscape(t *testing.T) {
+	should := require.New(t)
+
+	container := &Container{
+		Bar: []string{"123<ab>", "ooo"},
+	}
+	out, err := marshalConfig.Marshal(container)
+	should.Nil(err)
+	bufout := string(out)
+
+	var stdbuf bytes.Buffer
+	stdenc := json.NewEncoder(&stdbuf)
+	stdenc.SetEscapeHTML(false)
+	err = stdenc.Encode(container)
+	should.Nil(err)
+	stdout := string(stdbuf.Bytes())
+	if stdout[len(stdout)-1:] == "\n" {
+		stdout = stdout[:len(stdout)-1]
+	}
+
+	should.Equal(stdout, bufout)
+}

--- a/reflect_marshaler.go
+++ b/reflect_marshaler.go
@@ -3,8 +3,9 @@ package jsoniter
 import (
 	"encoding"
 	"encoding/json"
-	"github.com/modern-go/reflect2"
 	"unsafe"
+
+	"github.com/modern-go/reflect2"
 )
 
 var marshalerType = reflect2.TypeOfPtr((*json.Marshaler)(nil)).Elem()
@@ -93,10 +94,17 @@ func (encoder *marshalerEncoder) Encode(ptr unsafe.Pointer, stream *Stream) {
 		stream.WriteNil()
 		return
 	}
-	bytes, err := json.Marshal(obj)
+	marshaler := obj.(json.Marshaler)
+	bytes, err := marshaler.MarshalJSON()
 	if err != nil {
 		stream.Error = err
 	} else {
+		// html escape was already done by jsoniter
+		// but the extra '\n' should be trimed
+		l := len(bytes)
+		if l > 0 && bytes[l-1] == '\n' {
+			bytes = bytes[:l-1]
+		}
 		stream.Write(bytes)
 	}
 }

